### DISCUSSION TITLE
Unable to install openjdk@17 latest version on M1 mac

### DISCUSSION THREAD

**Discussion by alessandrolia** [Homebrew/discussions#5173] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8917acac9560f8e2846eedc4181470f0345f6db8
Last commit: 9 months ago
Core tap JSON: 31 May 13:42 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_AUTO_UPDATE: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.40.1 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 13.6.4-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
```

### Description of issue

I wanted to update my existing java 17 to 17.0.10, but when I run `brew install openjdk@17` I get:
```
Warning: openjdk@17 17.0.7 is already installed and up-to-date.
To reinstall 17.0.7, run:
  brew reinstall openjdk@17
```

But here I see there is a new version in the formula, to my very little understanding: https://github.com/Homebrew/homebrew-core/blob/cbe127f4cfeba403c79382ff2d8b982b123daf8c/Formula/o/openjdk@17.rb

Is that a bug or I'm missing something?
Thank a lot!

**Comment by ZhongRuoyu** [comment]
> HOMEBREW_VERSION: 4.0.20
> ORIGIN: https://github.com/Homebrew/brew
> HEAD: 8917acac9560f8e2846eedc4181470f0345f6db8
> Last commit: 9 months ago
> Core tap JSON: 31 May 13:42 UTC

Your Homebrew installation, local API cache, etc. are very outdated. Please do `brew update` first.

If you did not intentionally set `HOMEBREW_NO_AUTO_UPDATE`, you may want to unset that.

**Reply by p-linnane** [reply]
To add to this `HOMEBREW_NO_AUTO_UPDATE` prevents Homebrew from updating itself, not your installed packages.
>HOMEBREW_NO_AUTO_UPDATE
If set, do not automatically update before running some commands, e.g. brew install, brew upgrade and brew tap. Alternatively, run this less often by setting HOMEBREW_AUTO_UPDATE_SECS to a value higher than the default.

**Comment by alessandrolia** [comment]
Thanks a lot, after updating brew itself java has been updated correctly.
So noob, sorry

**Comment by alessandrolia** [comment]
Anyway, the update of brew also updatged EVERY package, forcing me to reconfigure some things... that's why I disabled it in thep ast (found a forgotten note in my zshrc)

**Reply by ZhongRuoyu** [reply]
`brew update` only updates `brew` itself, not the packages you install. `brew upgrade` does. So, if you don't do `brew upgrade`, nothing will be upgraded.

**Reply by alessandrolia** [reply]
I know this is the theory, anyway I just run `brew update` (checked twice in history) but it updated all my packages, forcing me to reconfigure many things. Fortunately it took me just an hour or 2, but quite annoying.
I don't know if I can inspect some logs to show you how it happened, but it did :(
No worries at all, it now works all fine
